protocols/mempool-overview.md

```diff
@@ -1,102 +1,80 @@
-
 # Mempool
 
-The mempool is a distributed pool of pending transactions.
-A pending transaction is a valid transaction that has been submitted by a
-client of the blockchain but has not yet been committed to the blockchain.
-The mempool is thus fed with client transactions,
-that a priori can be submitted to any node in the network.
-And it is consumed by the consensus protocol, more specifically by validator nodes,
-which retrieve from the mempool transactions to be included in proposed blocks.
-
-More concretely, every node participating in the mempool protocol maintains a
-local copy of the mempool, namely a list of pending transactions.
-Nodes that expose an interface to receive transactions from clients
-append the submitted transactions to their local copy of the mempool.
-These nodes are the entry point of the mempool protocol,
-and by extension of the consensus protocol.
-Nodes that play the role of validators in the consensus protocol,
-in their turn, retrieve from their local copy of the mempool
-pending transactions to be included in proposed blocks.
-Validator nodes are therefore the recipients of the transactions stored and
-transported by the mempool protocol.
-
-The goal of the mempool protocol is then to convey transactions
-from the nodes that act as entry points of the network
-to the nodes whose role is to order transactions.
+In this document, we define the notion of **mempool** and characterize its role in the **CometBFT** protocol.
+First, we provide an overview of what is a mempool, and relates it to other blockchains.
+Then, the interactions with the consensus and client application are detailed.
+A formalization of the mempool follows.
+We close this document with a proof that the current implementation of the mempool in CometBFT (as of v0.38.0-rc1) is correct.
+
+## Overview
+
+The mempool acts as an entry point to consensus.
+It permits to disseminate transactions from one node to another, for their eventual inclusion in the blockchain.
+To this end, the mempool maintains a replicated set, or _pool_, of transactions.
+Transactions in the mempool are consumed by consensus to create the next proposed block.
+The mempool is refreshed once a new block in the blockchain  is decided.
+
+A transaction can be received from a local client, or a remote disseminating process.
+Each transaction is subject to a test by the client application.
+This test verifies that the transaction is _valid_.
+Such a test provides some form of protection against byzantine agents, whether they be clients or other system nodes.
+It also serves to optimize the overall utility of the blockchain.
+Validity can be simply syntactical which is stateless, or a more complex verification that is state-dependent.
+If the transaction is valid, the local process further propagates it in the system using a gossip (or an anti-entropy) mechanism.
```

Comment:
The more I think, the less it appears that we are using a gossip protocol here, ehe.

Essentially, the dissemination is doing by flooding in a partially and randomly-connected network overlay.